AppImage/make-appimage.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+
+# This script is meant to be ran on alpine linux, dependencies:
+# wget build-base patchelf libxscrnsaver-dev libxinerama-dev libxrandr-dev libnotify-dev
+# dbus-dev wayland-dev perl pango-dev wayland-protocols dbus librsvg desktop-file-utils
+
+set -e
+export ARCH="$(uname -m)"
+export APPIMAGE_EXTRACT_AND_RUN=1
+APPIMAGETOOL="https://github.com/AppImage/appimagetool/releases/download/continuous/appimagetool-$ARCH.AppImage"
+export COMPLETIONS=0
+
+# BUILD DUNST
+mkdir -p ./AppDir
+make -j$(nproc) PREFIX="$PWD"/AppDir/usr
+make install PREFIX="$PWD"/AppDir/usr
+cd ./AppDir
+
+# AppRun
+echo "Creating AppRun..."
+cat >> ./AppRun << 'EOF'
+#!/bin/sh
+CURRENTDIR="$(dirname "$(readlink -f "$0")")"
+ARCH="$(uname -m)"
+export PATH="$CURRENTDIR/usr/bin:$PATH"
+[ -z "$APPIMAGE" ] && APPIMAGE="$0"
+BIN="${ARGV0#./}"
+unset ARGV0
+
+case "${BIN}" in
+	dunst|dunstify)
+		exec "$CURRENTDIR/ld-musl-${ARCH}.so.1" "$CURRENTDIR/usr/bin/$BIN" "$@"
+		;;
+	dunstctl)
+		exec "$CURRENTDIR/usr/bin/$BIN" "$@"
+		;;
+	""|*)
+		BIN="$1"
+		case "${BIN}" in
+			dunst|dunstify)
+				shift
+				exec "$CURRENTDIR/ld-musl-${ARCH}.so.1" "$CURRENTDIR/usr/bin/$BIN" "$@"
+				;;
+			dunstctl)
+				shift
+				exec "$CURRENTDIR/usr/bin/$BIN" "$@"
+				;;
+			""|*)
+				echo "AppImage commands:"
+				echo " \"$APPIMAGE dunst\"      runs dunst"
+				echo " \"$APPIMAGE dunstify\"   runs dunstify"
+				echo " \"$APPIMAGE dunstctl\"   runs dunstctl"
+				echo "You can also make and run symlinks to the AppImage with the names"
+				echo "dunstify and dunstctl to launch them automatically without extra args"
+				echo "running dunst..."
+				exec "$CURRENTDIR/ld-musl-${ARCH}.so.1" "$CURRENTDIR/usr/bin/dunst" "$@"
+				;;
+		esac
+esac
```

Comment:
Why is ther e a switch/case inside a switch/case now?

Reply:
Ah, now I get it. Was a bit confused by your overly long explanation. I'd probably put this in a for loop. But that's up to you at the end.

Reply:
I think it is fine as it is, I originally had an if statement that checked if `ARGV0` was a file in the appimage and if not it then checked `$1` and finally it would have just defaulted to running dunst, but this case statement is also ok.